### PR TITLE
Us003 view capability role belongs to

### PR DESCRIPTION
**Pull request by DarraghHaughey-Kainos**

### CODE REVIEW

File: src/main/java/org/kainos/ea/db/JobRoleDao.java [hunk 1]
```diff
@@ -3,25 +3,28 @@
 import org.kainos.ea.cli.JobRole;
 import org.kainos.ea.client.ActionFailedException;
 
-import java.sql.Connection;
-import java.sql.ResultSet;
-import java.sql.SQLException;
-import java.sql.Statement;
+import java.sql.*;
```
Comment (steveh2707) [inline src/main/java/org/kainos/ea/db/JobRoleDao.java]: Remove * import

**Review by steveh2707 — APPROVED** [review]
looks good - minor comment

File: src/main/java/org/kainos/ea/DropwizardWebServiceApplication.java [hunk 2]
```diff
@@ -33,5 +34,7 @@ public void run(final DropwizardWebServiceConfiguration configuration,
                     final Environment environment) {
         // TODO: implement application
         environment.jersey().register(new JobRoleController());
+        environment.jersey().register(new CapabilityController());
```
Comment (steveh2707) [inline src/main/java/org/kainos/ea/DropwizardWebServiceApplication.java]: instantiate service in this folder and pass into controller.

File: src/main/java/org/kainos/ea/cli/Capability.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
+package org.kainos.ea.cli;
+
+public class Capability {
+
+    private int capability_id;
+    private String capability_name;
```
Comment (steveh2707) [inline src/main/java/org/kainos/ea/cli/Capability.java]: camel case for these.

File: src/main/java/org/kainos/ea/cli/JobRole.java [hunk 4]
```diff
@@ -3,10 +3,12 @@
 public class JobRole {
     private String jobRoleTitle;
     private int id;
+    private String capability_name;
 
-    public JobRole(int id, String jobRoleTitle) {
+    public JobRole(int id, String jobRoleTitle, String capability_name) {
         this.jobRoleTitle = jobRoleTitle;
```
Comment (steveh2707) [inline src/main/java/org/kainos/ea/cli/JobRole.java:9]: camel case

File: src/main/java/org/kainos/ea/cli/JobRole.java [hunk 5]
```diff
@@ -24,4 +26,12 @@ public int getId() {
     public void setId(int id) {
         this.id = id;
     }
+
+    public String getCapability_name() {
```
Comment (steveh2707) [inline src/main/java/org/kainos/ea/cli/JobRole.java]: camel case

File: src/main/java/org/kainos/ea/db/JobRoleDao.java [hunk 6]
```diff
@@ -12,16 +12,22 @@
...
 public class JobRoleDao {
 
+    DatabaseConnector databaseConnector = new DatabaseConnector();
+
     public List<JobRole> getJobRoles(Connection c) throws ActionFailedException {
         try (Statement st = c.createStatement()) {
-            ResultSet rs = st.executeQuery("SELECT job_role_id, job_role_title FROM job_role;");
+            ResultSet rs = st.executeQuery("SELECT job_role_id, job_role_title, capability_name "
+                    + "FROM job_role "
+                    + "JOIN capability "
```
Comment (steveh2707) [inline src/main/java/org/kainos/ea/db/JobRoleDao.java]: should this be a left join?

File: src/main/java/org/kainos/ea/resources/CapabilityController.java [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+@Api("I Want It Data Way API")
+@Path("/api")
+public class CapabilityController {
+
+    private final CapabilityService capabilityService;
+
+    public CapabilityController() {
+        DatabaseConnector databaseConnector = new DatabaseConnector();
+        capabilityService = new CapabilityService(databaseConnector, new CapabilityDao());
+    }
```
Comment (steveh2707) [inline src/main/java/org/kainos/ea/resources/CapabilityController.java:22]: change constructor to pass in capabilityService

File: src/test/java/com/kainos/ea/integration/CapabilityControllerTest.java [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.kainos.ea.DropwizardWebServiceApplication;
+import org.kainos.ea.DropwizardWebServiceConfiguration;
+import org.kainos.ea.cli.Capability;
+
+import java.util.List;
+
+@ExtendWith(DropwizardExtensionsSupport.class)
+public class CapabilityControllerTest {
```
Comment (steveh2707) [inline src/test/java/com/kainos/ea/integration/CapabilityControllerTest.java]: rename to CapabilityIntegrationTest

**Review by steveh2707 — COMMENTED** [review]
Add controller tests

File: src/main/java/org/kainos/ea/cli/JobRole.java [hunk 9]
```diff
@@ -3,10 +3,12 @@
 public class JobRole {
     private String jobRoleTitle;
     private int id;
```
Comment (glenn-hamilton) [inline src/main/java/org/kainos/ea/cli/JobRole.java]: Could this ID be given a better name?

Comment (DarraghHaughey-Kainos) [reply src/main/java/org/kainos/ea/cli/JobRole.java]: changed it

**Review by glenn-hamilton — APPROVED** [review]

**Pull request merged**